### DISCUSSION TITLE
Dynamic Fields

### DISCUSSION THREAD

**Discussion by nassif7** [react-hook-form/react-hook-form#5649] [Q&A] [answered]
Hey there, 
I am trying to make a dynamic form, ie, the fields can be selected and rendered in the form. 
it works well when adding fields to the form. 
but when i am removing a field, the field is still appear in the formContext. 
do i need to unregister the fields in a way? 
i am using the Controller component.

here is the code. 

https://codesandbox.io/s/github/nassif7/Dynamic_React_Hook_Form

**Comment by bluebill1049** [comment, marked as answer]
Try to set shouldUnregister to true.

**Reply by nassif7** [reply]
many thanks, works very well
